src/test/mir-opt/inline/inline_single_private.main.Inline.diff

```diff
@@ -0,0 +1,111 @@
++                                          // + user_ty: UserType(0)
++                                          // + literal: Const { ty: fn(&[&'static str], &[std::fmt::ArgumentV1]) -> std::fmt::Arguments {std::fmt::Arguments::new_v1}, val: Value(Scalar(<ZST>)) }
+      }
+  
+      bb1: {
++         StorageDead(_2);                 // scope 0 at $DIR/inline-single-private.rs:22:25: 22:26
++         _0 = const ();                   // scope 0 at $DIR/inline-single-private.rs:20:11: 23:2
++         return;                          // scope 0 at $DIR/inline-single-private.rs:23:2: 23:2
++     }
++ 
++     bb2: {
++         StorageDead(_8);                 // scope 1 at $DIR/inline-single-private.rs:21:5: 21:17
++         StorageDead(_5);                 // scope 1 at $DIR/inline-single-private.rs:21:5: 21:17
++         _3 = _print(move _4) -> bb3;     // scope 1 at $DIR/inline-single-private.rs:21:5: 21:17
++                                          // mir::Constant
++                                          // + span: $DIR/inline-single-private.rs:21:5: 21:17
++                                          // + literal: Const { ty: for<'r> fn(std::fmt::Arguments<'r>) {std::io::_print}, val: Value(Scalar(<ZST>)) }
++     }
++ 
++     bb3: {
++         StorageDead(_4);                 // scope 1 at $DIR/inline-single-private.rs:21:5: 21:17
++         StorageDead(_10);                // scope 1 at $DIR/inline-single-private.rs:21:5: 21:17
++         StorageDead(_7);                 // scope 1 at $DIR/inline-single-private.rs:21:5: 21:17
++         StorageDead(_3);                 // scope 1 at $DIR/inline-single-private.rs:21:5: 21:17
++         StorageDead(_12);                // scope 0 at $DIR/inline-single-private.rs:21:5: 21:17
++         StorageDead(_11);                // scope 0 at $DIR/inline-single-private.rs:21:5: 21:17
+          StorageDead(_1);                 // scope 0 at $DIR/inline-single-private.rs:21:17: 21:18
+          StorageLive(_2);                 // scope 0 at $DIR/inline-single-private.rs:22:5: 22:25
+-         _2 = more_than_one_call() -> bb2; // scope 0 at $DIR/inline-single-private.rs:22:5: 22:25
++         _2 = more_than_one_call() -> bb1; // scope 0 at $DIR/inline-single-private.rs:22:5: 22:25
+                                           // mir::Constant
+                                           // + span: $DIR/inline-single-private.rs:22:5: 22:23
+                                           // + literal: Const { ty: fn() {more_than_one_call}, val: Value(Scalar(<ZST>)) }
+      }
++ }
+  
+-     bb2: {
+-         StorageDead(_2);                 // scope 0 at $DIR/inline-single-private.rs:22:25: 22:26
+-         _0 = const ();                   // scope 0 at $DIR/inline-single-private.rs:20:11: 23:2
+-         return;                          // scope 0 at $DIR/inline-single-private.rs:23:2: 23:2
+-     }
++ alloc5 (size: 0, align: 8) {}
```

Comment:
How can I exclude 32 bit targets from the test? This seems to be the reason test fails.

Reply:
You have two solutions:
1. add EMIT_MIR_FOR_EACH_BIT_WIDTH to the test and bless both 32bit and 64bit tests;
2. rewrite the contents of the functions to use regular instructions and no promoted constants (and adjust the threshold accordingly).